scrc/preprocessors/extractors/spider_specific/court_composition_extracting_functions.py

```diff
@@ -170,233 +170,315 @@ def prepare_french_name_and_find_gender(person: CourtPerson) -> CourtPerson:
+            CourtRole.CLERK: [r'Gerichtsschreiberin(nen)?', r'Gerichtssekretärin(nen)?', r'Sekretärin(nen)?']
         }
     }
 
-    information_start_regex = r'Mitwirkend:'
+    information_start_regex = r'Mitwirkend'
     start_pos = re.search(information_start_regex, header)
     if start_pos:
-        header = header[start_pos.span()[0]:]
+        # split off the first word
+        header = header[start_pos.span()[1]:]
 
     information_end_regex = r'In Sachen|in Sachen'
     end_pos = re.search(information_end_regex, header)
     if end_pos:
-        header = header[:end_pos.span()[1] - 1]
+        header = header[:end_pos.span()[1]]
+        # split off the last two words
+        header = header.rsplit(' ', 2)[0]
 
-    besetzung = CourtComposition()
-    besetzung = find_besetzung(header, role_regexes, namespace)
-    return besetzung
+    composition = CourtComposition()
+    composition = find_composition(header, role_regexes, namespace)
+    return composition
 
 
 
-def get_besetzungs_strings(header: str) -> list:
+def get_composition_strings(header: str) -> list:
+    """
+    Modifies the header of a decision and turns it into a list
+    :param header:  the header of a decision
+    :return:        a list of composition_strings
+    """
+    # repeating commas aren't necessary
+    header = re.sub(r', *, *', ', ', header)
+    # trying to join words that are split over two lines
+    header = re.sub(r'- *, *', '', header)
+    header = header.replace('- ', '')
+    # regularize different forms to denote the Vorsitz
+    header = header.replace('(Vorsitz)', 'Vorsitz')
+    header = header.replace('Vorsitzender', 'Vorsitz')
+    header = header.replace('Vorsitzende', 'Vorsitz')
+    header = header.replace('Vorsitz', ', Vorsitz, ')
+    # these word separators aren't relevant here
+    header = header.replace(':', '')
+    # a semicolon can be treated as a comma here
     header = header.replace(';', ',')
-    header = header.replace(' und ', ', ')
-    header = header.replace(' sowie ', ', ')
-    header = header.replace('lic. ', '')
-    header = header.replace('iur. ', '')
-    header = header.replace('Dr. ', '')
+    # der & die aren't relevant for this task
+    header = re.sub(r'\bder\b', '', header)
+    header = re.sub(r'\bdie\b', '', header)
+    # und & sowie separte different people
+    header = header.replace(' und', ', ')
+    header = header.replace(' sowie', ', ')
+    # academic degrees presumably aren't relevant for this task
+    header = header.replace('lic.', '')
+    header = header.replace('iur.', '')
+    header = header.replace('Dr.', '')
+    header = header.replace('Prof.', '')
     header = header.replace('MLaw ', '')
+    header = header.replace('M.A.', '')
+    header = header.replace('HSG ', '')
     header = header.replace('PD ', '')
+    header = header.replace('a.o.', '')
+    header = header.replace('LL.M.', '')
+    header = header.replace('LL. M.', '')
+    header = header.replace('LLM ', '')
```

Comment:
These you could for example combine into one with a regex using re.sub (https://docs.python.org/3/library/re.html#re.sub). But maybe it is more clear this way. Up to you.